xdsl/ir.py

```diff
@@ -665,6 +686,20 @@ def clone(self: OpT,
             region.clone_into(op.regions[idx], 0, value_mapper, block_mapper)
         return op
 
+    @classmethod
+    def has_trait(cls, trait: OpTrait) -> bool:
+        """
+        Check if the operation implements a trait with the given parameters.
+        """
+        return trait in cls.traits
+
+    @classmethod
+    def get_traits_of_type(cls, trait: type[OpTrait]) -> list[OpTrait]:
+        """
+        Get all the traits of the given type satisfied by this operation.
+        """
+        return [t for t in cls.traits if isinstance(t, trait)]
```

Comment:
Is there a chance for t not to be in trait?
If so, can this be checked at some earlier stage?

Reply:
trait here is a specific kind of trait, so something like NoResultsTrait. This will only return the traits that are also of that type.

Reply:
I'll rename this to `trait_type` or something, to make this more explicit.